### DISCUSSION TITLE
Non-gravitational forces from ACT1B when simulating an orbit

### DISCUSSION THREAD

**Discussion by ReubenLi1999** [groops-devs/groops#97] [Q&A] [answered]
Hello developers,

I realised that you guys have done a great job of simulating a pure dynamic orbit from a given force function which includes non-gravitational forces. 

It seems that the non-gravitational forces used here are calculated based on models, instead of measurements by the accelerometers. Have you guys considered calculating the non-conservative forces by the measurements of accelerometers?

It would be great if you add this feature. 

Best regard,

Reuben

**Comment by akvas** [comment]
Hi @ReubenLi1999,

thanks for this enhancement proposal! What would be your use case?

If you already have an approximate orbit, GROOPS can integrate dynamic orbits with accelerometer data. The program you are looking for is [PreprocessingVariationalEquation](https://groops-devs.github.io/groops/html/PreprocessingVariationalEquation.html). You can find a detailed description of how to use it in our [GRACE gravity field recovery cookbook](https://groops-devs.github.io/groops/html/cookbook.gravityFieldGrace.html). Would this be applicable to your use case?

Cheers,
Andreas

**Comment by ReubenLi1999** [comment]
Hi @akvas,

Thank you for the helpful guidance. I am sorry that I did not interpret my use case clearly. I want to propagate dynamic orbits of GRACE-FO from a given force model (static gravity and tides included) and ACT1B data, just like what you have done in [SimulateOrbit](https://groops-devs.github.io/groops/html/SimulateOrbit.html). The only difference is that ACT1B is required in my computation. Therefore, I am sorry that the content of [GRACE gravity filed recovery cookbook](https://groops-devs.github.io/groops/html/cookbook.gravityFieldGrace.html) does not seem to be what I want. 

Additionally, as you mentioned in [PreprocessingVariationalEquation](https://groops-devs.github.io/groops/html/PreprocessingVariationalEquation.html), non-conservative forces like solar radiation pressure need the orientation of the satellite ([inputfileStarCamera](https://groops-devs.github.io/groops/html/fileFormat_instrument.html)). However, I did not find any interface to input the orientation of the satellite (SCA) when [SimulateOrbit](https://groops-devs.github.io/groops/html/SimulateOrbit.html) is calculating the solar radiation pressure. Would you mind enlightening me on how to use the orientation of the satellite to calculate the non-gravitational forces?

Thanks for your time.

Best regard,

Reuben

**Comment by akvas** [comment, marked as answer]
Hi @ReubenLi1999,

I put together a [minimal example](https://github.com/groops-devs/groops/files/9382647/00groopsIntegrateOrbit.zip) which integrates an orbit for GRACE-C over a day.  Note that you will probably need to adapt the file paths for the L1B files and gravity field model. Let me know if that solves your problem.

Cheers,
Andreas

**Reply by ReubenLi1999** [reply]
Hi @akvas,

Thank you so much for the helpful guidance. The problem is solved based on the example you gave.

GROOPS is really a wondrous software!

Best regards,
Reuben
